src/Smalot/PdfParser/Document.php

```diff
@@ -263,12 +263,21 @@ public function extractXMPMetadata(string $content): void
                         break;
 
                     case 'close':
-                        // If the value of this property is a single-
-                        // element array where the element is of type
-                        // string, use the value of the first list item
-                        // as the value for this property
-                        if (\is_array($metadata) && isset($metadata[0]) && 1 == \count($metadata) && \is_string($metadata[0])) {
-                            $metadata = $metadata[0];
+                        // If the value of this property is an array
+                        if (\is_array($metadata)) {
+                            // If the value is a single element array
+                            // where the element is of type string, use
+                            // the value of the first list item as the
+                            // value for this property
+                            if (1 == \count($metadata) && isset($metadata[0]) && \is_string($metadata[0])) {
+                                $metadata = $metadata[0];
+
+                                // else if the value is an empty array, set
+                                // the value of this property to the empty
+                                // string
+                            } elseif (0 == \count($metadata)) {
+                                $metadata = '';
```

Comment:
```suggestion
                            } elseif (0 == \count($metadata)) {
                                // if the value is an empty array, set
                                // the value of this property to the empty
                                // string
                                $metadata = '';
```

Reply:
The only option left is that `$metadata` has more than 1 element. Why is that not possible here?

Reply:
> The only option left is that `$metadata` has more than 1 element. Why is that not possible here?

If `$metadata` has more than one element, it is not modified and remains an array. If it has zero or one elements (where the one element is of type `string`) then the value will be collapsed to a string.

For example, the following XMP XML:

```xml
<dc:Title>
  <rdf:Seq>
    <rdf:li></rdf:li>
  </rdf:Seq>
</dc:Title>
```

... would result in a value for `dc:title` of an empty string. However, for the following:

```xml
<dc:Title>
  <rdf:Seq>
  </rdf:Seq>
</dc:Title>
```

... the result was an empty array which is non-intuitive considering the behaviour of the first example.